bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkRemoteHandler.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    protected void sendCode(byte code[]) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        try {
+            byte[] preamble = new byte[4];
+            preamble[0] = 2;
+            outputStream.write(preamble);
+            outputStream.write(code);
+            byte[] padded = Utils.padTo(outputStream.toByteArray(), 16);
+            byte[] message = buildMessage((byte) 0x6a, padded);
+            sendAndReceiveDatagram(message, "remote code");
+        } catch (IOException e) {
+            logger.error("Exception while sending code", e);
+        }
+    }
+
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!Utils.isOnline(getThing())) {
+            logger.debug("Can't handle command {} because handler for thing {} is not ONLINE", command,
+                    getThing().getLabel());
+            return;
+        }
+        if (command instanceof RefreshType) {
+            updateItemStatus();
+            return;
+        }
+        Channel channel = thing.getChannel(channelUID.getId());
+        if (channel == null) {
+            logger.error("Unexpected null channel while handling command {}", command.toFullString());
+            return;
+        }
+        ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+        if (channelTypeUID == null) {
+            logger.error("Unexpected null channelTypeUID while handling command {}", command.toFullString());
+            return;
+        }
+        if (channelTypeUID.getId().equals("command")) {
+            logger.debug("Handling ir/rf command '{}' on channel {} of thing {}", command, channelUID.getId(),
+                    getThing().getLabel());
+            byte code[] = lookupCode(command, channelUID);
+            if (code != null) {
+                sendCode(code);
+            }
+        } else {
+            logger.debug("Thing {} has unknown channel type '{}'", getThing().getLabel(), channelTypeUID.getId());
+        }
+    }
+
+    @Nullable
+    private TransformationService getTransformService() {
+        // Lazy-load it
+        if (transformService == null) {
+            BundleContext bundleContext = FrameworkUtil.getBundle(BroadlinkRemoteHandler.class).getBundleContext();
+            transformService = TransformationHelper.getTransformationService(bundleContext, "MAP");
+            if (transformService == null) {
+                logger.error("Failed to get MAP transformation service for thing {}; is bundle installed?",
+                        getThing().getLabel());
+            }
+        }
+        return transformService;
+    }
+
+    private byte @Nullable [] lookupCode(Command command, ChannelUID channelUID) {
+        if (command.toString() == null) {
+            logger.debug("Unable to perform transform on null command string");
+            return null;
+        }
+        String mapFile = (String) thing.getConfiguration().get("mapFilename");
```

Comment:
```suggestion
        String mapFile = thing.thingConfig.getMapFilename();
```
If this works I believe it will be more efficient.